src/compositor/compositor.c

```diff
@@ -1077,11 +1076,23 @@ static void gst_imx_video_compositor_update_overall_region(GstImxVideoCompositor
 	while (walk != NULL)
 	{
 		GstImxVideoCompositorPad *compositor_pad = GST_IMX_VIDEO_COMPOSITOR_PAD_CAST(walk->data);
-		GstImxRegion *outer_region = &(compositor_pad->canvas.outer_region);
+		GstImxRegion outer_region = compositor_pad->canvas.outer_region;
 		GstVideoInfo *info = &(GST_IMXBP_VIDEO_AGGREGATOR_PAD(compositor_pad)->info);
 
+		/* Flip width/height if a transposing rotation is done
+		 * FIXME: Is this correct? Does this take all the
+		 * transformations into account, e.g. should
+		 * gst_imx_canvas_calculate_inner_region() be used instead?
```

Comment:
The above is basically the remaining question. It works for my use-case, but maybe it would give wrong results if other transformations are configured on the pad.